src/endpoint/s3/ops/s3_get_bucket.js

```diff
@@ -30,7 +30,18 @@ async function get_bucket(req) {
             (cont_tok_to_key_marker(cont_tok) || start_after) : req.query.marker,
     };
 
+    if (req.query.get_from_cache) {
+        params.get_from_cache = req.query.get_from_cache;
+    }
     const reply = await req.object_sdk.list_objects(params);
+
+    // Update the object creation time in case this is from cache
+    if (req.query.get_from_cache) {
+        _.forEach(reply.objects, obj => {
+            obj.create_time = obj.last_modified_time ? obj.last_modified_time : obj.create_time;
```

Comment:
You know, we don't really want to modify these objects properties meaning,
why not put this inside the response building in line 69 like so:
```
                    // if the object specifies last_modified_time we use it, otherwise take create_time.
                    LastModified: s3_utils.format_s3_xml_date(obj.last_modified_time || obj.create_time),
```